packages/connect/src/protocol-connect/transport.ts

```diff
@@ -78,6 +80,7 @@ export function createTransport(opt: CommonTransportOptions): Transport {
         opt.jsonOptions,
         opt
       );
+      const ac = createLinkedAbortController(signal);
```

Comment:
This is the relevant change: We create an AbortSignal and pass it to the HTTP client, and we wrap the rest of the Transport logic in a try-catch block and abort the signal on error. For HTTP/2, this closes the stream (and session), for HTTP/1.1 it calls destroy() on request and response.